semanticcpg/src/main/scala/io/shiftleft/semanticcpg/codedumper/CodeDumper.scala

```diff
@@ -0,0 +1,76 @@
+package io.shiftleft.semanticcpg.codedumper
+
+import io.shiftleft.codepropertygraph.generated.{NodeTypes, nodes}
+import io.shiftleft.semanticcpg.language._
+import io.shiftleft.semanticcpg.language.NodeSteps
+import better.files._
+import org.apache.logging.log4j.LogManager
+
+import scala.util.Try
+
+object CodeDumper {
+
+  private val logger = LogManager.getLogger(CodeDumper)
+  val arrow: CharSequence = "// ===>\n"
+
+  /**
+    * Evaluate the `step` and determine associated locations.
+    * Dump code at those locations
+    * */
+  def dump[NodeType <: nodes.StoredNode](step: NodeSteps[NodeType]): String =
+    step.location.l.map(dump).mkString("\n")
+
+  /**
+    * Dump string representation of code at given `location`.
+    * */
+  def dump(location: nodes.NewLocation): String = {
+    val filename = location.filename
+
+    if (location.node.isEmpty) {
+      logger.warn("Empty `location.node` encountered")
+      return ""
+    }
+
+    val node = location.node.get
+    val method = node match {
+      case n: nodes.Method     => Some(n)
+      case n: nodes.Expression => Some(n.method)
+      case _                   => None
+    }
+
+    val lineToHighlight = location.lineNumber
+    method
+      .filter { m =>
+        m.lineNumber.isDefined && m.lineNumberEnd.isDefined
+      }
+      .map { m =>
+        code(filename, m.lineNumber.get, m.lineNumberEnd.get, lineToHighlight)
+      }
+      .getOrElse("")
```

Comment:
Depending on the input size in comparison to the amount of data filtered, `collect` might be faster here as it does not create two intermediate lists. Other than that, its a matter of personal preference.
(see: https://stackoverflow.com/questions/36961777/scala-can-there-be-any-reason-to-prefer-filtermap-over-collect)